pkg/client/clientset.go

```diff
@@ -152,8 +153,15 @@ func CreatePlatformClientSet(url string, auth PlatformAuth, opts ClientOptions)
 		automation.WithCustomUserAgentString(opts.getUserAgentString()),
 	)
 
-	bucketClient := bucket.NewClient(url, platformClient)
-
+	bucketClient, err := clients.Factory().
+		WithOAuthCredentials(clientcredentials.Config{
+			ClientID:     auth.OauthClientID,
+			ClientSecret: auth.OauthClientSecret,
+			TokenURL:     auth.OauthTokenURL,
+		}).
+		WithEnvironmentURL(url).
+		WithUserAgent(opts.getUserAgentString()).
+		BucketClient()
 	if err != nil {
```

Comment:
good catch by the linter above. 
The dtclient creation error from line 136 is never checked anymore.
Before it was checked by this err != nil, which now gets the error from the clients.Factory build.